internal/builders/go/pkg/provenance.go

```diff
@@ -70,6 +70,11 @@ func GenerateProvenance(name, digest, command, envs, workingDir string, s signin
 		return nil, err
 	}
 
+	// If the environment does not have access to an OIDC provider, use a nil one.
+	if provider == nil && !github.HasOIDCClient() {
```

Comment:
are there cases where provider is not `nil` and has not OIDC client? Existing code suggest no.